dvc/cli.py

```diff
@@ -102,11 +139,48 @@ def _find_parser(parser, cmd_cls):
 class DvcParser(argparse.ArgumentParser):
     """Custom parser class for dvc CLI."""
 
+    cmd_choices = {}
+    hidden_cmds = ["completion", "daemon", "exp", "experiments", "git-hook"]
```

Comment:
Doesn't seem like `completion` should be hidden though. Also, the rest of the commands are already marked as hidden (e.g. `add_help=False` in dvc/command/daemon.py). Would it be possible to just detect that while we parse `cmd_choices`? Would make maintaining this easier :slightly_smiling_face:

Reply:
Definitely. Will dig into more of the `argparse` internals to see if we can detect hidden commands.